### ISSUE TITLE
Data Discrepancies 2021

### ISSUE THREAD

**Issue by CaymanTurtleBeach** [binance/binance-public-data#103]
There appear to be significant discrepancies in historic data for Binance 15m BTCUSDT Kline futures (and although I have not specifically confirmed, I assume that lower and higher resolution klines are similarly affected).

Data downloaded from data.binance.vision/?prefix=data/futures/um/monthly/klies/BTCUSDT/15m

March 2 2021 (2021-03-02 01:00:00.000Z) the number of trades and volume differ greatly from the prior kline (ending at 00:45:59.999Z). Then the next 3 klines all have no price changes and zero volume (the final ignored field is not shown here).
**00:45** kline 1614645900000 49681.98 49782.22 49310 49401.01 4229.44 1614646799999 209252269.9 **39210** 1920.517 95032171.34
**01:00** kline 1614646800000 49401 49413.38 49361 49361.01 63.932 1614647699999 3157740.221 **457** 28.1 1388152.24
**01:15** kline 1614647700000 49361.01 49361.01 49361.01 49361.01 0 1614648599999 **0 0 0 0**
(01:30 and 01:45 are identical to 01:15 kline other than the time stamps)

for 1 January 2021, data from the historic file does not fully match either the kline retrieved from
**fapi/v1/kline_15m** OR **fapi/v2/continuousKline_15m** (PERPETUAL contract type):
File Data (organized to correspond to API return):

	[
		1609459200000,
		28948.19,
		29045.93,
		28706.00,
		28786.75,
		3181.960,
		1609460099999,
		91834596.81370,
		24166,
		1277.389,
		36899008.19096
		0
		]
**fapi/v1/kline_15m** data:

    [
        1609459200000,
        "28948.19",
        "29045.93",
        "28706.00",
        "28786.75",
        "42236.270",
        1609460099999,
        "1220393434.49536",
        1895,
        "19324.997",
        "558571428.81492",
        "0"
    ]
**fapi/v1/continuousKline_15m** data:

    [
        1609459200000,
        "28907.76",
        "29011.21",
        "28697.39",
        "28750.21",
        "42236.270",
        1609460099999,
        "1220393434.49536",
        1895,
        "19324.997",
        "558571428.81492",
        "0"
    ]
Prices from historic file match Prices from fapi/v1/kline_15m but **Volume** does **NOT**.

Volume from **fapi/v1/kline_15m** and **fapi/v1/continuousKline_15m** matches but **Prices** do **NOT**.
These glaring data discrepancies at the least do not bolster confidence in Binance data integrity, and without data integrity everything else is also suspect.
I do note that there appears to be consistency with current klines (fapi/v1/kline_15m and fapi/v1/continuous kline_15m, retrieved today December 25).
Questions:

Will these issues (e.g., the zero values in March data) be fixed?
Is the URL I used for historic data (manual downloading zip files by month) correct?
What is the difference between Futures klines (fapi/v1/kline_*X*) and fapi/v1/continuousKline_*X* with a contract type of PERPETUAL?
On the face of it, it seems to me they are/should be the same.

**Comment by 2pd** [comment]
- continuousKline_X consist of kline_X
- For quarterly delivery contract, the symbol will be listed/delivered every three months, and kline_X data will stop update if delivered, so we build continuousKline_X to link them.

**Comment by volkan-akkus-tr** [comment]
Hi @CaymanTurtleBeach ,
did you solve your suspicions on data consistency ?
Regards,

**Comment by CaymanTurtleBeach** [comment]
@volkan-akkus-tr I am not sure I would say my suspicions are fully solved with respect to Klines, although the answer from Binance suggests that they took action and presumably Kline data is now consistent. After I found issues with Klines, I decided to use the raw AggTrade data that is generated monthly. For the periods I have looked carefully at (OCT 2021 through FEB 2022) the monthly AggTrade data seems to be completely consistent. Order IDs monotonically increase sequentially as expected, and the transitions from the end of one month the beginning of the next month show no time gaps.
Cheers,
